data-processing-lib/python/src/data_processing/data_access/data_access_s3.py

```diff
@@ -51,19 +51,17 @@ def __init__(
         self.s3_credentials = {} | s3_credentials
         access_key = self.get_access_key()
         if access_key is None:
+            logger.warning("S3 access key not provided. Defaulting to ''")
```

Comment:
I do not see much point in this. If the access key is not provided, we should exit (throw an exception). Data Access factory do just this in validation

Reply:
done

Reply:
If the S3 data access key is not provided, IMHO, the program should use the credentials given in the `~/.aws/credentials` file.  This is the default behavior of the `boto3` client.

Reply:
Sorry, no. We tried it before and it led to a lot of confusion. Also, remember, our main goal is running on the cluster not locally